### PR TITLE
Ab/cb30/create layout l008 requirements

### PR DESCRIPTION
**Pull request by Andresbl123**
Implement the `Requirements` component to improve the way requirements are visualised. Key changes include:

this section will show the requirements mock to implement the layout, which will be implemented. 

Updated the `Fieldset` component to include a ‘Requirements’ heading.

Mapped through the data passed to `Requirements` to generate multiple `TableBoard` components. Each `TableBoard` receives a unique set of `titles`, `entries` and `actions`.

image
image

### CODE REVIEW

File: src/pages/board/outlets/boardlayout/Requirements/index.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+export interface IRequirementsProps {
+  data: IData[];
+}
+export const Requirements = (props: IRequirementsProps) => {
+  const { data } = props;
+
+  return (
+    <Stack>
+      <Fieldset title="Requerimientos">
```
Comment (novadeveloper14) [inline src/pages/board/outlets/boardlayout/Requirements/index.tsx:22]: should say "Requisitos":

image

**Review by novadeveloper14 — CHANGES_REQUESTED** [review]
The component must have the "Add requirements" button. according to what is seen in the requirements of the task in Jira

image

**Review by carin91 — CHANGES_REQUESTED** [review]

File: src/pages/board/outlets/boardlayout/Requirements/index.tsx (hunk 1 above)
Comment (Andresbl123) [reply src/pages/board/outlets/boardlayout/Requirements/index.tsx:22]: resuelto

File: src/pages/board/outlets/boardlayout/Requirements/config.tsx [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+            cursorHover
+          />
+        </div>
+      ),
+    },
+    {
+      id: "aprobar",
+      actionName: "Aprobar",
+      content: (data: IEntries) => (
+        <div style={{ display: "flex", justifyContent: "center" }}>
```
[the remark this answers is not in the document]
Comment (Andresbl123) [reply src/pages/board/outlets/boardlayout/Requirements/config.tsx]: resuelto

File: src/pages/board/outlets/boardlayout/Requirements/config.tsx [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+  ],
+];
+
+export const actionsRequirements = [
+  [
+    {
+      id: "agregar",
+      actionName: "Agregar",
+      content: (data: IEntries) => (
+        <div style={{ display: "flex", justifyContent: "center" }}>
```
[the remark this answers is not in the document]
Comment (Andresbl123) [reply src/pages/board/outlets/boardlayout/Requirements/config.tsx]: resuelto

File: src/App.tsx [hunk 4]
```diff
@@ -36,7 +38,14 @@ const router = createBrowserRouter(
       <Route path="login/*" element={<LoginRoutes />} />
       <Route path="/*" element={<BoardRoutes />} />
       <Route path="logout" element={<LogOut />} />
-      <Route path="financial-reporting" element={<FinancialReporting />} />
+      <Route
```
[the remark this answers is not in the document]
Comment (Andresbl123) [reply src/App.tsx:41]: resuelto

File: src/pages/board/outlets/boardlayout/Requirements/index.tsx [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+}
+
+export interface IRequirementsProps {
+  data: IData[];
+}
+export const Requirements = (props: IRequirementsProps) => {
+  const { data } = props;
+
+  return (
+    <Stack>
```
[the remark this answers is not in the document]
Comment (Andresbl123) [reply src/pages/board/outlets/boardlayout/Requirements/index.tsx:22]: resuelto

**Review by novadeveloper14 — APPROVED** [review]

File: src/pages/board/outlets/boardlayout/Requirements/index.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (carin91) [reply src/pages/board/outlets/boardlayout/Requirements/index.tsx:22]: The layout rendered seems invalid.

Validate the gap of the (+) and (check) buttons, as the location of the tags.

image

File: src/App.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (carin91) [reply src/App.tsx:41]: This task is pending

File: src/pages/board/outlets/boardlayout/Requirements/index.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (Andresbl123) [reply src/pages/board/outlets/boardlayout/Requirements/index.tsx:22]: resuelto

File: src/App.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (Andresbl123) [reply src/App.tsx]: resuelto

**Review by carin91 — APPROVED** [review]

**Pull request merged**